app/services/katello/candlepin/message_handler.rb

```diff
@@ -85,6 +89,36 @@ def delete_pool
           Rails.logger.info "deleted pool #{pool_id} from Katello"
         end
       end
+
+      def handle_content_access_mode_modified
+        # Ideally the target_name would be the Candlepin Owner key
+        # Since it's the displayName, and we don't update that after org creation, there could be a mismatch
+        # For now, find the Candlepin Owner displayName from this event, and tie it back to a Katello org based on owner key
+        owners = Katello::Resources::Candlepin::Owner.all
+        owner = owners.find { |o| o['displayName'] == target_name }
+
+        unless owner
+          fail("Candlepin Owner %s could not be found" % target_name)
+        end
+
+        org = ::Organization.find_by!(label: owner['key'])
+        hosts = org.hosts
+
+        if event_data['contentAccessMode'] == 'org_environment'
+          Katello::HostStatusManager.clear_syspurpose_status(hosts)
+          Katello::HostStatusManager.update_subscription_status_to_sca(hosts)
+        elsif event_data['contentAccessMode'] == 'entitlement'
+          cp_consumer_uuids = hosts.joins(:subscription_facet).pluck("#{Katello::Host::SubscriptionFacet.table_name}.uuid")
+          cp_consumer_uuids.each do |uuid|
+            Katello::Resources::Candlepin::Consumer.compliance(uuid)
+            Katello::Resources::Candlepin::Consumer.purpose_compliance(uuid)
+          rescue => e
+            Rails.logger.error("Error encountered while fetching compliance for consumer #{uuid}: #{e.message}")
+          end
+        end
+
+        org.simple_content_access?(cached: false)
```

Comment:
Seems like this return value isn't used anywhere; is it necessary?

Reply:
Here I'm refreshing our cached value for `simple_content_access?` so that any request which reads it 'soon' (the cache has a 1 min lifetime) will be accurate. We cache it because it's a frequent call that we make, and saves requests into Candlepin. Just a coincidence that it's the last line of the method!